src/Build/Instance/TaskFactories/AssemblyTaskFactory.cs

```diff
@@ -68,6 +69,8 @@ internal class AssemblyTaskFactory : ITaskFactory2
         /// </summary>
         private TaskLoggingContext _taskLoggingContext;
 
+        private bool _isTaskHostFactory;
```

Comment:
this business with `_taskHostFactoryExplicitlyRequested`, `_isTaskHostFactory` has confusing naming. Can it be improved, or at least comment the field.